nodeJS/express/views.md

```diff
@@ -78,9 +78,29 @@ Start the server and go to the `/` route in the browser. You should see:
 EJS rocks!
 ```
 
-If you inspect the HTML in the browser's dev tools, you can see HTML is structures exactly like how we wrote the EJS template with the `message` variable replaced with its value.
+If you inspect the HTML in the browser's dev tools, you can see the HTML is structured exactly like how we wrote the EJS template with the `message` variable replaced with its value.
 
-When you hit the `/` route, `res.render("index", { message: "EJS rocks!" });` is the line that sends back the response. Since we've already defined the `views` and `view engine` app properties, the first argument of `res.render` is programmed to look for "a template called index in the specified folder", while the second argument is an object of variables that are to be made available to that specific template.
+When you hit the `/` route, `res.render("index", { message: "EJS rocks!" });` is the line that sends back the response. Since we've already defined the `views` and `view engine` app properties, the first argument of `res.render` is programmed to look for "a template called index in the specified folder", while the second argument is an object of variables that are to be made available to that specific template. This object of variables is known as the `locals` object which is always generated by EJS. Say you define a variable called `foo` in the `locals` object. This variable can be accessed in the view file with either `locals.foo` or `foo`. A few more details can be found in the [Express docs](https://expressjs.com/en/5x/api.html#res.locals).
+
+In the example above, the `message` property exists on the `locals` object as `locals.message`, to which we can access with either `locals.message` or more directly with `message`.
+
+<div class="lesson-note lesson-note--tip" markdown="1">
+
+#### Undefined variables in locals
+
+If we try to access a variable in a rendered template file that was not defined in `res.locals`, this can cause a reference error. For instance if we try to access an undefined `foo` variable, `locals.foo` will return undefined, while `foo` will result in an reference error. Verify this by adding `locals.foo`, and then `foo` below `message` in `index.ejs`:
```

Comment:
Not just `res.locals` but also if not passed in via the 2nd argument in `res.render`.

Reply:
I'll change this to include both the `locals` argument of `res.render`, and `res.locals`.